### ISSUE TITLE
[BUG] OpenDialog with accept and chrome not working

### ISSUE THREAD

**Issue by Paul6552** [react-dropzone/react-dropzone#1190]
**Describe the bug**
When you have set the attribute accept the file dialog does not open in chrome.

**To Reproduce**
Go to "Accepting specific file types": [Click me](https://react-dropzone.js.org/#section-accepting-specific-file-types)
Try to click on the dropzone-field "Drag and drop some files here or click to select"
The file dialog does not open in chrome. I tested Edge and firefox, these browsers are working.

When you remove the accept attribute, the file dialog does also open in chrome.

**Expected behavior**
File dialog should open

Versions:
Chrome latest
Linux Mint
nodeJs latest
react-dropzone 14.2.0

**Comment by clovisp** [comment]
Hello, I'm having a similar issue, but not exactly the same.

On my computer, the file dialog does open, however I can select only the first media type set in the object passed to accept argument.
For instance if we have this object: 
```
useDropzone({
  accept: {
    'image/png': ['.png'],
    'text/html': ['.html', '.htm'],
  }
})
```

Then only png images are selectable. Html files remain greyed out.

Versions:
On chrome (Version 101.0.4951.54) and macOS Monterey

**Comment by Paul6552** [comment]
@clovisp 
Different bug. Open a new one ;-)

**Comment by rolandjitsu** [comment]
@Paul6552 please add the `onError` cb and log to see if you get any errors (or check the console for any errors as it defaults to that):
```
useDropzone({
  ...
  onError: e => console.log(e)
})
```

Do note that if you're not in a [secure context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts), the [File System Access API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API) is supported and  you do not provide the `input`, the file dialog cannot be opened.

**Comment by rolandjitsu** [comment]
@clovisp please check https://github.com/react-dropzone/react-dropzone/issues/1191.

**Comment by Paul6552** [comment]
@rolandjitsu 
I am in a secure context.

I added the onError and do not see any errors :-/

Not sure If you have seen, but the link I posted in my first post is from your website. 

https://user-images.githubusercontent.com/25934591/167241418-8303433b-104c-4d99-9b39-6387428295e1.mp4

**Comment by rolandjitsu** [comment]
This could be an issue with Linux only. Could you try the following:
```js
async function getFile() {
  // open file picker
  try {
    const [fileHandle] = await window.showOpenFilePicker();
    const canRead = verifyPermission(fileHandle);

    if (canRead) {
      const fileData = await fileHandle.getFile();
      console.log(fileData);
    } else {
      console.warn('no permissions to read file')
    }
  } catch (e) {
    console.error(e)
  }
}

async function verifyPermission(fileHandle, withWrite) {
  const opts = {};
  // Check if we already have permission, if so, return true.
  if (await fileHandle.queryPermission(opts) === 'granted') {
    return true;
  }

  // Request permission to the file, if the user grants permission, return true.
  if (await fileHandle.requestPermission(opts) === 'granted') {
    return true;
  }

  // The user did not grant permission, return false.
  return false;
}

getFile()
```

**Comment by Paul6552** [comment]
After running this I get the FileDialog from Linux.

![image](https://user-images.githubusercontent.com/25934591/167246480-38300b5b-1281-4eee-9861-3dfe1228ca26.png)

**Comment by rolandjitsu** [comment]
> After running this I get the FileDialog from Linux.

Do you mean that if you run the code I shared it works, as in, it opens the file picker?

> Up from 13.0.0 FileDialog is not opening anymore

That's when we made the FS access API the default way to open the file picker.

@Paul6552 does the example sandbox work for you ([jq6d5](https://codesandbox.io/s/react-dropzone-example-jq6d5?file=/src/App.js))?

**Comment by Paul6552** [comment]
> Do you mean that if you run the code I shared it works, as in, it opens the file picker?

Yes

The example you provided also works. I am now more confused then before.

First I tried your example as it is, with React 18. It works.
I modified your example and changed it to react 17.0.2. It works.
Then I pasted the code with my code (identical to your website's code) and it works.

I don't understand why it doesn't work on your react-dropzone website and on my website.

**Comment by rolandjitsu** [comment]
@Paul6552 ok.

> The example you provided also works. I am now more confused then before.
> First I tried your example as it is, with React 18. It works.
> I modified your example and changed it to react 17.0.2. It works.

I see. So everything works as expected.

> I forked again the ["create react app"](https://github.com/facebook/create-react-app) and inserted only your Dropzone code (from your website).
> Again. File Opener does not open.

Would you mind sharing that? Also, if you test [react-dropzone-test](https://github.com/rolandjitsu/react-dropzone-test) as is, does it work or not?

> I added useFsAccessApi: false. After that everything is working fine.

That means the issue is related to the FS access API. It could be just a permissions thing. you could try to request permissions before the dropzone is used.

> I have no more attempt to solve the problem, so I'll stick with this attitude.
> Should I close the ticket or should it stay open?

Let's keep open until we track down the issue.

> Maybe File System Access Api shouldnt be activated by default. It is still experimental, so why should we use an experimental feature in production

As of now, that API is the only way to get a reliable cancel callback when the user closes the file picker dialog without picking any files. Using the input is also not a conventional way of triggering the file picker dialog to open (not the way we use it in the lib), it's a workaround since there's no other APIs.

On that note, please checkout https://github.com/react-dropzone/react-dropzone/discussions/1186 and vote.

**Comment by Paul6552** [comment]
> Would you mind sharing that? Also, if you test [react-dropzone-test](https://github.com/rolandjitsu/react-dropzone-test) as is, does it work or not?

Right now I am too stupid to create a new repository to github and share the link to you. Have never done it before. 
What have I done:
1. Fork the [create react app](https://github.com/facebook/create-react-app) from facebook
2. Add react-dropzone 14.2.1 in the package.json
3. Copy exactly that code from your website: [Example from react-dropzone](https://react-dropzone.js.org/#!/Accepting%20specific%20file%20types/1)
4. npm install and npm start

I tested "react-dropzone-test" as well. Same behavior, FileDialog does not open.

> On that note, please checkout #1186 and vote.

I voted

**Comment by johnhunter** [comment]
@clovisp your/our issue is due to poor UX in the MacOS file picker. See the screen recording in #1191 for the aha moment 🙂

**Comment by ml242** [comment]
> > Do you mean that if you run the code I shared it works, as in, it opens the file picker?
> 
> Yes
> 
> The example you provided also works. I am now more confused then before.
> 
> First I tried your example as it is, with React 18. It works. I modified your example and changed it to react 17.0.2. It works. Then I pasted the code with my code (identical to your website's code) and it works.
> 
> I don't understand why it doesn't work on your react-dropzone website and on my website.
> 
> //Edit I forked again the ["create react app"](https://github.com/facebook/create-react-app) and inserted only your Dropzone code (from your website). Again. File Opener does not open.
> 
> //Edit2: I added useFsAccessApi: false. After that everything is working fine.
> 
> I have no more attempt to solve the problem, so I'll stick with this attitude. Should I close the ticket or should it stay open?
> 
> //Edit3 Maybe File System Access Api shouldnt be activated by default. It is still experimental, so why should we use an experimental feature in production

suggestion 3 is very reasonable after having the same issue.

**Comment by chrismou** [comment]
Sounds like you guys have it nailed, but to confirm this is also happening to me using an `accept` and affects Chrome and Edge on Linux for me. Firefox still seems to work OK

I've also voted :slightly_smiling_face:

**Comment by flobby** [comment]
Can confirm, any updatest?

**Comment by temrb** [comment]
+1

accept: {
'video/* ': ['.hevc', '.mp4', '.mov', '.m4v', '.avi'],
'image/* ': ['.jpeg', '.png', '.heif'],
},

for me it doesn't accept image if it's video first. Doesn't accept video if image is first in the object.

**Comment by kuzniets** [comment]
+1

same behaviour, any news? There was a new version yesterday but without a fix for the current issue (

**Comment by rolandjitsu** [comment]
> @rolandjitsu The problem seems to be due to a [bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1264708#c10) with Chromium on Linux. Providing a `description` seems to workaround the issue.

Hmm. That's interesting. Does an empty description work?

**Comment by EkaanshArora** [comment]
Can reproduce with Chrome on Linux. An empty description doesn't work for me, any other string fixes it.
Any ideas on adding a workaround? The Chromium team hasn't prioritized this bug. Happy to work on a PR.

**Comment by j-avila** [comment]
I'm having the same issue on chrome (macOS),  the select file dialog doesn't open. In some point tried the `useFsAccessApi: false`  and works for some but suddenly stop working,  any workaround o solution for this chrome bug?

**Comment by getalihassan** [comment]
Ubuntu 22.04
react-dropzone: 14.2.2
chrome: 104.0.5112.101
ran into same issue

useFsAccessApi: false

fixed it for me.
just wondering will it have any effects on macOS, windows?

**Comment by stalet** [comment]
+1 same behaviour.
Used the workaround `useFsAccessApi: false` to fix it.

**Comment by mdsavian** [comment]
I have the same issue in Linux, using the `useFsAccessApi: false` seems work fine.

**Comment by dlech** [comment]
This sounds like the same issue as #1127. It seems like a browser bug in the file system access API where omitting the supposedly optional description for the file types that are populated by `access` causes the dialog to be canceled without ever being shown to the user. This is why not using `access` or disabling use of the file system access api works around the issue. I filed https://crbug.com/1359728 for this.

**Comment by nmay231** [comment]
@dlech It was already filed: https://bugs.chromium.org/p/chromium/issues/detail?id=1264708#c10

**Comment by dlech** [comment]
:+1: Thanks (I did search first, really :smile:).

**Comment by mgtitimoli** [comment]
Hello everyone,

I have just experienced this issue using chrome in Linux, I tested the workaround of passing `useFsAccessApi: false` and it worked.

I also tested calling [window.showOpenFilePicker](https://developer.mozilla.org/en-US/docs/Web/API/window/showOpenFilePicker) and it didn't work when I passed no description.

Couldn't we pass a description generating from all the file extensions concatenated with ", " [here](https://github.com/react-dropzone/react-dropzone/blob/a2039fd4bc5430a166858d71b7499a17036e68f9/src/index.js#L768) to fix this issue?

In case we all agree to do that I can submit a PR with the fix, just let me know. 

Thanks.

**Comment by dlech** [comment]
There is already PR #1233.

**Comment by mgtitimoli** [comment]
> There is already PR #1233.

What I'm proposing is a bit different, instead of hardcoding "Files" in the description, I was thinking on receiving a new param so the user could pass the thing she wants and in case none was given, get a default one by concatenating all the accepted files extensions as it does the native file picker, something as follows:

```ts
const getFilePickerDescription = (acceptedFileTypes: Accepted) =>
  Object.values(acceptedFileTypes).flat().join(", ");
```

**Comment by godinhojoao** [comment]
Hi, just trying to help who needs.

I was having this same problem and I tried to change my react-dropzone version from 14.2.2 to 11.4.2 and other older versions, but it didn't work.

I also tried to use the property: `useFsAccessApi: false`, but it didn't work too.

So I decided to solve this problem on my own. As I already knew that the problem was ocurring because of the property `accept: { 'image/jpeg': [], 'image/png': [] }`, I did remove this property and then I created my expected validation into the `onDrop` function.

Example:
```javascript
 async function uploadImage (currentFile) {
    const acceptedFormats = ['image/jpeg', 'image/png', 'image/jpg']
    const file = currentFile && currentFile[0]
  
    const isFileValid = file && acceptedFormats.includes(file.type)
    if (!isFileValid) { // return error  }
    
    // do upload
 }
```
